pkg/config/viper.go

```diff
@@ -96,7 +96,10 @@ func (c *safeConfig) IsSectionSet(section string) bool {
 	// This is needed when keys within the section
 	// are set through env variables.
 	for _, key := range c.AllKeys() {
-		if strings.HasPrefix(key, section) && c.IsSet(key) {
+		// Add trailing . to make sure we don't take into account unrelated
+		// settings, eg. IsSectionSet("section") shouldn't return true
+		// if "section_key" is set.
+		if strings.HasPrefix(key, section+".") && c.IsSet(key) {
```

Comment:
Would it make sense to move `section+"."` outside of the loop to avoid allocating a new copy on every iteration?

Reply:
Yes, I can do that 👍